src/map/anticheat.h

```diff
@@ -0,0 +1,86 @@
+/*
+===========================================================================
+
+  Copyright (c) 2010-2019 Darkstar Dev Teams
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see http://www.gnu.org/licenses/
+
+  This file is part of DarkStar-server source code.
+
+===========================================================================
+*/
+
+#ifndef _CANTICHEAT_H
+#define _CANTICHEAT_H
+
+#include "../common/cbasetypes.h"
+#include "entities/charentity.h"
+
+namespace anticheat
+{
+    // Cheat type as appears in the cheat id column in the cheat_incidents table.
+    enum class CheatID : uint32
+    {
+        // Player attempted to speed up synth (craft) by injecting a synth done
+        // (0x59) packet. If anti cheating is enabled this will cause all cheated
+        // crafts to fail.
+        // Argument is the time spent on synth animation.
+        CHEAT_ID_FASTSYNTH = 1,
+        // For boundary checks
+        CHEAT_ID_LAST
+    };
+
+    enum CheatActionBitmask
```

Comment:
enum class

Reply:
I'd rather not use enum class for bitmasks, causes the code to become uglier when doing bitwise operation. I only use enum because C++ doesn't really have a proper bitmask type that I know of (I think that using #define in CPP is even worse).
Regarding the second comment, changed (will push soon).

Reply:
Pushed.